### DISCUSSION TITLE
AWS CLI not allowing spaces in string passed to flag value

### DISCUSSION THREAD

**Discussion by Ali-Khalifa98** [aws/aws-cli#9108] [Q&A]
I am trying to execute aws cli command with flag --description "Hello World" but it fails because of the white space.

example command am trying:

` aws lexv2-models update-intent --description "Hello World"`

the error I receive

` Unknown options: World"`

**Comment by uldyssian-sh** [comment]
This is not an AWS CLI limitation — it is a shell parsing issue.

The AWS CLI fully supports spaces in flag values, but the shell must pass the value as a single argument. The error you see means the shell is splitting the string before it reaches AWS CLI.

Use the quoting rules appropriate for your shell.

On Bash / zsh (Linux, macOS, WSL):

aws lexv2-models update-intent --description "Hello World"

or equivalently:

aws lexv2-models update-intent --description 'Hello World'


On Windows PowerShell (recommended):

aws lexv2-models update-intent --description "Hello World"

or:

aws lexv2-models update-intent --description 'Hello World'


On Windows CMD, make sure you are using straight quotes (not smart quotes) and not escaping incorrectly:

aws lexv2-models update-intent --description "Hello World"


If the command still fails, double-check that:
- the quotes are ASCII quotes (`"`), not typographic quotes
- the command is not being executed through another layer (script, CI, npm, etc.) that strips or reinterprets quotes

In short: AWS CLI accepts spaces just fine — the issue is always how the shell passes the argument, not the CLI itself.

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
